### PR TITLE
[SCB-2238] FSM event channel supports Rabbit

### PR DESCRIPTION
**Pull request by githubcheng2978**

### CODE REVIEW

**Comment by githubcheng2978** [comment]
#686

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/rabbit/RabbitChannelAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+@EnableConfigurationProperties(BindingServiceProperties.class)
+@ConditionalOnProperty(value = "alpha.feature.akka.channel.type", havingValue = "rabbit")
+public class RabbitChannelAutoConfiguration {
+
+
+    @Bean
+    @ConditionalOnMissingBean()
+    public RabbitMessagePublisher rabbitMessagePublisher(BindingServiceProperties bindingServiceProperties, RabbitMessageChannel producerMessage) {
+        Map<String, BindingProperties> bindings = bindingServiceProperties.getBindings();
+        // 分区数量,现在现在生产者与消费这都在alpha-server，所以rabbit的分区partitionCount与该数量保持一直
```
Comment (WillemJiang) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/rabbit/RabbitChannelAutoConfiguration.java:48]: Please add English comments.

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/rabbit/RabbitChannelAutoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        int partitionCount = bindings.get(RabbitMessageChannel.SERVICE_COMB_PACK_PRODUCER).getProducer().getPartitionCount();
+        RabbitMessagePublisher messagePublisher = new RabbitMessagePublisher(partitionCount, producerMessage);
+        return messagePublisher;
+    }
+
+//    @StreamMessageConverter
```
Comment (WillemJiang) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/rabbit/RabbitChannelAutoConfiguration.java:54]: Please remove the commented code.

File: alpha/alpha-server/src/main/resources/application.yaml [hunk 3]
```diff
@@ -146,6 +146,46 @@ spring:
...
+  profiles: rabbit
+  cloud:
+    stream:
+      binders:
+        defaultRabbit:
+          type: rabbit
+          environment:
+            spring:
+              rabbitmq:
+                host: 127.0.0.1
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/resources/application.yaml:160]: It's better to provide the default setting as docker compose does.

**Review by WillemJiang — COMMENTED** [review]
Thanks for the contribution, we also need to update the accept test to do an end to end verification.

**Comment by githubcheng2978** [comment]
@WillemJiang  麻烦看看我这个PR哪里还有问题呢

**Comment by coolbeevip** [comment]
I trigger CI run via reopening this issue

**Comment by githubcheng2978** [comment]
> I trigger CI run via reopening this issue

so what should i do to resolve this problem

**Comment by coolbeevip** [comment]
> > I trigger CI run via reopening this issue
> 
> so what should i do to resolve this problem

We just migrated CI from Travis to Github Action, your PR has been pass CI.

You need to modify the code according to William's suggestion and push it to [githubcheng2978:feature-rabbitmq](https://github.com/githubcheng2978/servicecomb-pack/tree/feature-rabbitmq) again

**Comment by githubcheng2978** [comment]
> William's

**Comment by githubcheng2978** [comment]
> > > I trigger CI run via reopening this issue
> > 
> > 
> > so what should i do to resolve this problem
> 
> We just migrated CI from Travis to Github Action, your PR has been pass CI.
> 
> You need to modify the code according to William's suggestion and push it to [githubcheng2978:feature-rabbitmq](https://github.com/githubcheng2978/servicecomb-pack/tree/feature-rabbitmq) again

 ok , thanks for you help.  i  already finished it

**Comment by githubcheng2978** [comment]
@coolbeevip 什么时候可以merge 啊

**Comment by WillemJiang** [comment]
> @WillemJiang 麻烦看看我这个PR哪里还有问题呢

我看代码你把rabbitmq的客户端代码引入到Alpha server 的环境中来了，如果用户没有使用rabbitmq的话，是不是会报系统连接不上的信息。为了避免这个副作用，建议把RabbitMQ的配置文件从main/resources 去除。 
另外如果能附上验收测试就更好了。

File: alpha/alpha-server/src/main/resources/application.yaml (hunk 3 above)
Comment (coolbeevip) [reply alpha/alpha-server/src/main/resources/application.yaml:160]: Please remove the rabbit default configuration in the application.yml file and add rabbit configuration description to the document https://github.com/apache/servicecomb-pack/blob/master/docs/fsm/eventchannel_zh.md

**Comment by coolbeevip** [comment]
I created a JIRA https://issues.apache.org/jira/browse/SCB-2238 for this PR

**Comment by coolbeevip** [comment]
> @coolbeevip 什么时候可以merge 啊

方便的话，你可以给我发个邮件 coolbeevip@gmail.com, 我们交换一下微信

File: alpha/alpha-server/src/main/resources/application.yaml (hunk 3 above)
Comment (githubcheng2978) [reply alpha/alpha-server/src/main/resources/application.yaml]: done

**Review by coolbeevip — APPROVED** [review]

**Comment by githubcheng2978** [comment]
> > @WillemJiang 麻烦看看我这个PR哪里还有问题呢
> 
> 我看代码你把rabbitmq的客户端代码引入到Alpha server 的环境中来了，如果用户没有使用rabbitmq的话，是不是会报系统连接不上的信息。为了避免这个副作用，建议把RabbitMQ的配置文件从main/resources 去除。
> 另外如果能附上验收测试就更好了。

已经把RabbitMQ的配置文件从main/resources 移除掉了。

**Review by WillemJiang — APPROVED** [review]

**Pull request merged**